spacemesh/v1/admin_types.proto

```diff
@@ -15,3 +17,84 @@ message RecoverRequest {
   string uri = 1;
   uint32 restore_layer = 2;
 }
+
+message Event {
+  // time when event occured.
+  google.protobuf.Timestamp timestamp = 1;
+  // the reason of the failure may be complex to pinpoint.
+  // for the first version we want to highlight that failure occured and defer user to logs. 
+  bool failure = 2;
+  string help = 3;
+  oneof details {
+    EventBeacon beacon = 4;
+    EventInitStart init_start = 5;
+    EventInitComplete init_complete = 6;
+    EventPostStart post_start = 7;
+    EventPostComplete post_complete = 8;
+    EventPoetWait poet_wait = 9;
+    EventPoetWaitChallenge poet_wait_challenge = 10;
+    EventAtxPubished atx_published = 11;
+    EventEligibilities eligibilities = 12;
+    EventProposal proposal = 13;
+  }
+}
+
+message EventBeacon {
+  uint32 epoch = 1;
+  bytes beacon = 2;
+}
+
+message EventInitStart {
+  bytes smesher = 1;
+  bytes commitment = 2;
+}
+
+message EventInitComplete {}
+
+message EventPostStart {
+  bytes challenge = 1;
+}
+
+message EventPostComplete {
+  bytes challenge = 1;
+}
+
+message EventPoetWait {
+  uint32 current = 1;
+  uint32 publish = 2;
+  google.protobuf.Duration wait = 3;
+}
+
+message EventPoetWaitChallenge {
+  uint32 publish = 1;
+  uint32 target = 2;
+  google.protobuf.Duration wait = 3;
+}
+
+message EventAtxPubished {
+  uint32 current = 1;
+  uint32 target = 2;
+  bytes id = 3;
+  google.protobuf.Duration wait = 4;
+}
+
+message EventEligibilities {
+  uint32 epoch = 1;
+  bytes beacon = 2;
+  bytes atx = 3;
+  uint32 active_set_size = 4;
+  repeated ProposalEligibility eligibilities = 5;
+}
+
+message ProposalEligibility {
+  uint32 layer = 1;
+  uint32 count = 2;
+}
+
+// Published proposal. Rewards will be received, once proposal is included into the block.
+message EventProposal {
+  uint32 layer = 1;
+  bytes proposal = 2;
```

Comment:
do we not need to include weight or number of eligibilities here? to aid in calculating expected reward

Reply:
rewards are distributed based on relative weight in the block, i don't think that estimating makes any sense